src/main/java/Parser.java

```diff
@@ -0,0 +1,135 @@
+                System.out.println("    Oops! The description of a todo cannot be empty.");
+            } finally {
+                break;
+            }
+        case "deadline":
+            try {
+                String[] details = commands[1].split(" /by ", 2);
+                Task deadline = new Deadline(details[0], details[1]);
+                list.add(deadline);
+                printAdded(deadline, list);
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println("    Oops! A deadline must contain both a task "
+                        + "description and a deadline.");
+            } finally {
+                break;
+            }
+        case "event":
+            try {
+                String[] details = commands[1].split(" /at ", 2);
+                Task event = new Event(details[0], details[1]);
+                list.add(event);
+                printAdded(event, list);
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println("    Oops! An event must contain both an event "
+                        + "description and event time.");
+            } finally {
+                break;
+            }
+        case "done":
+            try {
+                int index = Integer.parseInt(commands[1]) - 1;
+                list.get(index).markDone();
+                System.out.println("    Nice! I've marked this task as done:");
+                System.out.println("      " + list.get(index).getTaskDisplay());
+            } catch (IndexOutOfBoundsException e) {
+                printIndexException();
+            } catch (NumberFormatException e) {
+                printNumFormatException();
+            } finally {
+                break;
+            }
+        case "delete":
+            try {
+                int index = Integer.parseInt(commands[1]) - 1;
+                Task toDelete = list.remove(index);
+                System.out.println("    Noted. I've removed this task:");
+                System.out.println("      " + toDelete.getTaskDisplay());
+                System.out.println("    Now you have " + list.size() + " task"
+                        + ((list.size() == 1) ? "" : "s") + " in the list.");
+            } catch (IndexOutOfBoundsException e) {
+                printIndexException();
+            } catch (NumberFormatException e) {
+                printNumFormatException();
+            } finally {
+                break;
+            }
+        case "find":
+            try {
+                System.out.println("    Here are the matching tasks in your list:");
+                int listSize = list.size();
+                for (int i = 0; i < listSize; i++) {
+                    if (list.get(i).getName().contains(commands[1])) {
+                        System.out.println("    " + (i + 1) + "." + list.get(i).getTaskDisplay());
+                    }
+                }
+            } catch (IndexOutOfBoundsException e) {
```

Comment:
here the possible exception is when commands[1] is null, which does not triggers index out of bound exception,
should change to null pointer exception.